pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/creator/BitSlicedIndexCreatorTest.java

```diff
@@ -250,24 +276,37 @@ private void testDouble(Dataset<double[]> dataset)
     File rangeIndexFile = new File(INDEX_DIR, metadata.getColumnName() + BITMAP_RANGE_INDEX_FILE_EXTENSION);
     try (PinotDataBuffer dataBuffer = PinotDataBuffer.mapReadOnlyBigEndianFile(rangeIndexFile)) {
       BitSlicedRangeIndexReader reader = new BitSlicedRangeIndexReader(dataBuffer, metadata);
-      double prev = Double.NEGATIVE_INFINITY;
-      for (double quantile : dataset.quantiles()) {
-        ImmutableRoaringBitmap reference = dataset.scan(prev, quantile);
-        ImmutableRoaringBitmap result = reader.getMatchingDocIds(prev, quantile);
-        int resultCount = reader.getNumMatchingDocs(prev, quantile);
-        assertEquals(result, reference);
-        assertEquals(resultCount, result == null ? 0 : result.getCardinality());
+      double[] quantiles = dataset.quantiles();
+      // TODO: RangeBitmap has a bug in version 0.9.28 which gives wrong result computing between for 2 doubles with
+      //       different sign. The bug is tracked here: https://github.com/RoaringBitmap/RoaringBitmap/issues/586.
+      //       Uncomment this line after the bug is fixed.
+//      double prev = quantiles[0] - 1;
```

Comment:
A temporary workaround is provided on the issue, I suggest applying the suggestion in this PR and revisit this when the bug is fixed (though that should be soon).

Reply:
Sounds good